code/weapon/swarm.cpp

```diff
@@ -128,65 +117,36 @@ void swarm_maybe_fire_missile(int shipnum)
 	}
 }
 
+// if it can't home on something, just set target dead ahead
+void swarm_set_defaut_target_pos(object* objp, swarm_info* swarmp) {
+	vm_vec_scale_add(&swarmp->current_target, &objp->pos, &objp->orient.vec.fvec, SWARM_CONE_LENGTH);
+}
+
 // ------------------------------------------------------------------
 // swarm_create()
 //
-//	Get a free swarm missile entry, and initialize the struct members
+//	Set up all the basic info for a swarm missile
 //
-int swarm_create()
+void swarm_create(object* objp, swarm_info* swarmp)
 {
-	int			i;
-	swarm_info	*swarmp = NULL;
-
-	for ( i = 0; i < MAX_SWARM_MISSILES; i++ ) {
-		swarmp = &Swarm_missiles[i];
-		if ( !(swarmp->flags & SWARM_USED) ) 
-			break;		
-	}
-
-	if ( i >= MAX_SWARM_MISSILES ) {
-		nprintf(("Warning","No more swarm missiles are available\n"));
-		return -1;
-	}
-
-	swarmp->flags = 0;
 	swarmp->change_timestamp = 1;
-	swarmp->path_num = -1;
 	swarmp->homing_objnum = -1;
+	swarmp->zig_direction = golden_ratio_rand() * PI2;
```

Comment:
This is a big ask, but is there a way to create a static version of this rand function for multi syncing?  Anything based on the net signature will really help the missiles to take the same path.  If anyone decides later on to create a swarm bomb that the multiplayer clients have to shoot down, having the static function will make it possible rather than completely impossible.

Reply:
Yeah, that shouldn't be a problem.

Reply:
Thank you!